libdnf/utils/File.cpp

```diff
@@ -0,0 +1,65 @@
+#include "File.hpp"
+#include "CompressedFile.hpp"
+#include "utils.hpp"
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <sstream>
+#include <utility>
+#include <iostream>
+
+extern "C" {
+#   include <solv/solv_xfopen.h>
+};
+
+std::shared_ptr<libdnf::File> libdnf::File::getFile(std::string filePath)
```

Comment:
Who needs shared_ptr? The most expensive solution.
getFile() in fact is something like newFile because it creates new instance of File class (or child of File).